Applications/MaterialAssist/Contracts/Boards/Interfaces/IMaterialAssistClientBoards.cs

```diff
@@ -0,0 +1,171 @@
+        Task DeleteBoardEntity(string id);
+
+        /// <summary>
+        /// Delete a board entities from the inventory by their ids (#).
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <returns></returns>
+        Task DeleteBoardEntities(IEnumerable<string> ids);
+
+
+        #endregion Delete
+
+        #region Read
+
+        /// <summary>
+        /// Get all board entities from the inventory.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntities(int take, int skip = 0);
+
+        /// <summary>
+        /// Get board entity by id (#).
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        Task<BoardEntity> GetBoardEntityById(string id);
+
+        /// <summary>
+        /// Get board entities by id´s (#).
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByIds(IEnumerable<string> ids);
+
+        /// <summary>
+        /// Get board entities by board code.
+        /// </summary>
+        /// <param name="boardCode"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCode(string boardCode);
+
+        /// <summary>
+        /// Get board entities by board codes.
+        /// </summary>
+        /// <param name="boardCodes"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCodes(IEnumerable<string> boardCodes);
+
+        /// <summary>
+        /// Get board entities by material code.
+        /// </summary>
+        /// <param name="materialCode"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByMaterialCode(string materialCode);
+
+        /// <summary>
+        /// Get board entities by material codes.
+        /// </summary>
+        /// <param name="materialCodes"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByMaterialCodes(IEnumerable<string> materialCodes);
+
+        /// <summary>
+        /// Get storage locations.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<StorageLocation>> GetStorageLocations();
+
+        /// <summary>
+        /// Get storage locations by workstation.
+        /// </summary>
+        /// <param name="workstation"></param>
+        /// <returns></returns>
+        Task<IEnumerable<StorageLocation>> GetStorageLocations(string workstation);
+
+
+        #endregion Read
+
+        #region Create
+
+        /// <summary>
+        /// Create a board instance for a certain boardType.
+        /// </summary>
+        /// <param name="boardTypeCode"></param>
+        /// <param name="boardEntity"></param>
+        /// <returns></returns>
+        Task<BoardEntity> CreateBoardEntity(string boardTypeCode, BoardEntity boardEntity);
```

Comment:
Also here. Use a request object. We should be consistent.

Reply:
this is a concept, we don´t have the request object yet